app.js

```diff
@@ -271,15 +283,32 @@ const completeRecoveryValidation = async ctx => {
 
     const account = await models.Account.findOne({ where: { accountId } });
 
+    if (!account) {
+        ctx.throw(401, 'account does not exist');
+    }
+
     const [recoveryMethod] = await account.getRecoveryMethods({ where: {
         kind: method.kind,
         detail: method.detail,
         securityCode: securityCode
     }});
 
     if (!recoveryMethod) {
-        ctx.throw(401);
+        ctx.throw(401, 'recoveryMethod does not exist');
+    }
+
+    // for new accounts, clear all other recovery methods that may have been created
+    if (ctx.request.url.indexOf('ForTempAccount') > -1) {
```

Comment:
@mattlockyer note that the way this check is done means I can make request to `/initializeRecoveryMethod?thisPartIsIgnored=ForTempAccount` and so effectively bypass a check that account doesn't exist before.

Reply:
There are 2 safe ways to do this:
1) set whatever option you need on `ctx.` in `checkAccountDoesNotExist`
2) have a wrapper function which would take it, so that usage of middleware would be smth like:
```
router.post('/account/validateSecurityCodeForTempAccount',
    checkAccountDoesNotExist,
     completeRecoveryValidation({ isNew: true })
 );
```

Reply:
Great catch.

Using solution 2.